storage/blob_test.go

```diff
@@ -389,6 +390,41 @@ func (s *StorageBlobSuite) TestGetBlobRange(c *chk.C) {
 	}
 }
 
+func (s *StorageBlobSuite) TestCreateBlockBlobFromReader(c *chk.C) {
+	cli := getBlobClient(c)
+	cnt := randContainer()
+	c.Assert(cli.CreateContainer(cnt, ContainerAccessTypePrivate), chk.IsNil)
+	defer cli.deleteContainer(cnt)
+
+	name := randString(20)
+	data := randBytes(8888)
+	c.Assert(cli.CreateBlockBlobFromReader(cnt, name, uint64(len(data)), bytes.NewReader(data)), chk.IsNil)
+
+	body, err := cli.GetBlob(cnt, name)
+	c.Assert(err, chk.IsNil)
+	gotData, err := ioutil.ReadAll(body)
+	body.Close()
+
+	c.Assert(err, chk.IsNil)
+	c.Assert(gotData, chk.DeepEquals, data)
+}
+
+func (s *StorageBlobSuite) TestCreateBlockBlobFromReaderWithShortData(c *chk.C) {
+	cli := getBlobClient(c)
+	cnt := randContainer()
+	c.Assert(cli.CreateContainer(cnt, ContainerAccessTypePrivate), chk.IsNil)
+	defer cli.deleteContainer(cnt)
+
+	name := randString(20)
+	data := randBytes(8888)
+	err := cli.CreateBlockBlobFromReader(cnt, name, 9999, bytes.NewReader(data))
```

Comment:
Yes, up to this line the test is just like the previous one. But the previous one only tests a "success" case. This one tests whether an upload error ("we said we were uploading a 9999-byte blob, but we provided fewer than 9999 bytes of data") causes `CreateBlock` to return an error (so our application knows something went wrong) and avoids storing the truncated data in the container (so we don't have to wonder every time our application calls `GetBlob` whether we're getting back bogus data due to a previous upload mishap).

Perhaps `...WithTruncatedData` would be a better hint than `...WithShortData` about what's being tested?